### PR TITLE
#321 support deploy to subfolder or reverse proxy via nginx

### PR DESCRIPTION
**Pull request by iflamed**
these update support view it via http://domain.com/editor/ and http://domain.com/editor/index.html; but it can't be visited by  http://domain.com/editor;

my nginx config is like blow:
```nginx
location /editor {
         proxy_pass http://localhost:9000/;
}
```

### CODE REVIEW

**Comment by iflamed** [comment]
@mohsen1 if the travis_ci build is fixed, please tell me and i will post a new pr.

File: app/scripts/bootstrap.js [hunk 1]
```diff
@@ -9,11 +9,17 @@ $(function () {
     window.SwaggerEditor.$defaults = embeddedDefaults;
     angular.bootstrap(window.document, ['SwaggerEditor']);
   } else {
-    $.getJSON('/config/defaults.json').done(function (resp) {
+    var rootPath = '';
+    if (window.location.pathname.lastIndexOf('/') !==
```
Comment (mohsen1) [inline app/scripts/bootstrap.js]: Could you explain what's going on here?

File: app/scripts/controllers/main.js [hunk 2]
```diff
@@ -43,7 +43,12 @@ SwaggerEditor.controller('MainCtrl', function MainCtrl($rootScope, $stateParams,
 
       // If there is no saved YAML either, load the default example
       } else if (!yaml) {
-        url = defaults.examplesFolder + defaults.exampleFiles[0];
+        var rootPath = '';
```
Comment (mohsen1) [inline app/scripts/controllers/main.js]: Could you explain what's going on here also?

**Comment by mohsen1** [comment]
Please rebase from master. It will pass tests if you do.

File: app/scripts/bootstrap.js (hunk 1 above)
Comment (iflamed) [inline app/scripts/bootstrap.js]:
```javascript
$.getJSON('/config/defaults.json')
```
This make a request base on the url root. if we deploy the dist folder to sub path like http://domain.com/editor/index.html. Then it get the defaults.json by the url http://domain.com/config/defaults.json so throw a 404. 

the window.location.pathname would be end with '/' or 'html' or a dirname like 'editor'. When it end with '/', it dosen't need to add a root path,just change the '/config/defaults.json' to './config/defaults.json' will be ok. When it end with 'html' or a dirname like 'editor' , the root path is the parent directory, so rootPath = pathname + '/../' . This logic fixes the defaults.json and the examples yaml 404 error, when deploy it in a sub folder or reverse proxy by nginx.

File: app/scripts/controllers/main.js (hunk 2 above)
Comment (iflamed) [inline app/scripts/controllers/main.js]: the same as above.

**Comment by mohsen1** [comment]
:+1:

**Pull request merged**